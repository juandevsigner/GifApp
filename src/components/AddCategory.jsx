import { useState } from "react";
import Proptypes from "prop-types";

export const AddCategory = ({ onNewCategory }) => {
  const [inputValue, setInputValue] = useState("");

  const onInputChange = ({ target }) => {
    setInputValue(target.value);
  };

  const onSubmit = e => {
    e.preventDefault();
    if (inputValue.trim().length <= 1) return;
    onNewCategory(inputValue.trim());
    setInputValue("");
  };

  return (
    <form aria-label="form" onSubmit={onSubmit}>
      <input
        type="text"
        value={inputValue}
        onChange={onInputChange}
        placeholder="Search Gifs"
      />
    </form>
  );
};

AddCategory.proptypes = {
  onNewCategory: Proptypes.func.isRequired,
};
