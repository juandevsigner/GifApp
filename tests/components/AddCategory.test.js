import { screen, render, fireEvent } from "@testing-library/react"
import { AddCategory } from "../../src/components/AddCategory"

describe('AddCategory test', () => {
    it('Snapshot vs', () =>{
        const {container} = render(<AddCategory onNewCategory={() => {}} />)
        expect(container).toMatchSnapshot()
    })
    it('should change input value', () =>{
        render(<AddCategory onNewCategory={() => {}} />)
        const input = screen.getByRole('textbox')
        fireEvent.input(input, {target: {value:'Saitama'}})
        expect(input.value).toBe('Saitama')
    })
    it('should call onNewCategory if input has value', ()=>{
        const inputValue = 'Saitama'
        const onNewCategory = jest.fn()
        render(<AddCategory onNewCategory={onNewCategory} />)
        const input = screen.getByRole('textbox')
        const form = screen.getByRole('form')
        fireEvent.input(input, {target: {value:inputValue}})
        fireEvent.submit(form)
        expect(input.value).toBe('')
        expect(onNewCategory).toHaveBeenCalledTimes(1)
        expect(onNewCategory).toHaveBeenCalledWith(inputValue)
    })
    it('should not call onNewCategory if input has no value', ()=>{
        const inputValue = ''
        const onNewCategory = jest.fn()
        render(<AddCategory onNewCategory={onNewCategory} />)
        const input = screen.getByRole('textbox')
        const form = screen.getByRole('form')
        fireEvent.input(input, {target: {value:inputValue}})
        fireEvent.submit(form)
        expect(input.value).toBe('')
        expect(onNewCategory).not.toHaveBeenCalled()
    })
})