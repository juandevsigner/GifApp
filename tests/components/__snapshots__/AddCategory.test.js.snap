// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddCategory test Snapshot vs 1`] = `
<div>
  <form
    aria-label="form"
  >
    <input
      placeholder="Search Gifs"
      type="text"
      value=""
    />
  </form>
</div>
`;
